.circleci/config.yml

```diff
@@ -0,0 +1,116 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: zimbra/zm-base-os:devcore-ubuntu-16.04
+    environment:
+      - ANT_OPTS: "-Dzimbra.buildinfo.version=8.9.0"
+      - BUILD: "/home/build"
+      - CONTAINERNAME: zmc-mailbox
+    working_directory: /home/build/zm-mailbox-circleci
+    # branches:
+    #   only:
+    #     - circleci
+    user: build
+    steps:
+      - checkout
+      - run:
+          name: Create build directory
+          command: mkdir -p "$BUILD"
+
+      # Build the submodules
+      - run:
+          name: Building 'zm-mailbox/native'
+          command: ant -f native/build.xml compile publish-local
+      - run:
+          name: Building 'zm-mailbox/common'
+          command: ant -f common/build.xml compile publish-local
+      - run:
+          name: Building 'zm-mailbox/soap'
+          command: ant -f soap/build.xml compile publish-local
+      - run:
+          name: Building 'zm-mailbox/client'
+          command: ant -f client/build.xml compile publish-local
+      - run:
+          name: Building 'zm-mailbox/store'
+          command: ant -f store/build.xml compile publish-local
+
+      # Run the unit tests separately for each project
+      - run:
+          name: Running unit tests for 'zm-mailbox/native'
+          command: ant -f native/build.xml test
+      - run:
+          name: Running unit tests for 'zm-mailbox/common'
+          command: ant -f common/build.xml test
+      - run:
+          name: Running unit tests for 'zm-mailbox/soap'
+          command: ant -f soap/build.xml test
+      - run:
+          name: Running unit tests for 'zm-mailbox/client'
+          command: ant -f client/build.xml test
+      - run:
+          name: Running unit tests for 'zm-mailbox/store'
+          command: ant -f store/build.xml test
+
+      - run:
+          name: Preparing to build installer - checking out zm-build & zm-docker
+          command: |
+            mkdir -p "$BUILD"
+            cd "$BUILD"
+            git clone --depth=1 --branch feature/ha git@github.com:Zimbra/zm-build.git zm-build
+            git clone --depth=1 --branch develop git@github.com:Zimbra/zm-docker.git zm-docker
```

Comment:
Can we avoid use of zm-docker all together?
Zm-mailbox should contain entry point script from mailbox service. 
Same for config.build

Reply:
Yes that can be done.
This image is intended to actually be run inside of `zm-docker` at this time and so this task will be undertaken in the future.

Reply:
i.e. they are tightly coupled and using the entry scripts from zm-docker is necessary to enable it to run as expected in that environment.